#!/usr/bin/env python
 #  -*-coding:utf8 -*-
PACKAGE = "base_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#修改之后要重新编译

#轮子转5圈，用皮尺测量距离70cm=700mm,700/5=140mm
gen.add("wheel_circumference_mm", int_t,    0, "An Integer parameter", 140,  0,  1000) 

#1430->1424 #轮子转5圈，然后除以5，LU7144/7101,RU7104/7138,LD-7121/-7101,RD-7139/-7103 -->1424
#L7154,R7109,L-7094,R-7122 --> 1424
gen.add("pulses_per_wheel_turn_around",    int_t,    0, "An Integer parameter", 1424,  0, 10000) 

#打开雷达，放在地面上，逆转5圈，顺转5圈
#N:-11527 11515,  -11552 11558 --> 2304, 2311 --> 2308
#S:11464 -11483,  11488 -11490 --> 2295, 2298 --> 2297
gen.add("pulses_per_robot_turn_around",    int_t,    0, "An Integer parameter", 2303,  0, 10000) #2331-->2299-->2303

gen.add("max_pwm",                  int_t,    0, "An Integer parameter", 6400,  0, 10000)
gen.add("pid_p",                int_t,    0, "An Integer parameter", 50,  0,  200)
gen.add("pid_i",                int_t,    0, "An Integer parameter", 20,  0,  200)
gen.add("pid_d",                int_t,    0, "An Integer parameter", 20,  0,  200)

exit(gen.generate(PACKAGE, "base_control", "Base"))

